Raven.Voron/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -110,19 +97,20 @@ public virtual void MaybePrefetchMemory(List<long> pagesToPrefetch)
             // do nothing
         }
 
-        public abstract byte* AcquirePagePointer(long pageNumber, PagerState pagerState = null);
-
-        public abstract void Sync();
-
-        public virtual PagerState TransactionBegan()
+        public virtual byte* AcquirePagePointer(Transaction tx, long pageNumber, PagerState pagerState = null)
         {
             ThrowObjectDisposedIfNeeded();
 
-            var state = PagerState;
-            state.AddRef();
-            return state;
+            var state = pagerState ?? PagerState;
+
+            if (tx != null)
+                tx.EnsurePagerStateReference(state);
+
```

Comment:
Here is the crucial change. Even if we read from the current `PagerState` we need to ensure it's properly referenced by a transaction so it won't be released if the transaction is still active.